### DISCUSSION TITLE
Complains "Track is closed"  but then works

### DISCUSSION THREAD

**Discussion by wusspuss** [paullouisageneau/libdatachannel#995] [Q&A]
I'm trying to accept a session and send audio from the microphone, with the C api. The flow I have:
1. `setRemoteDescription`
2. `onGatheringStateChange` when COMPLETE ->`rtcGetLocalDescription` and send to the peer
3.  `rtcTrackCallback` -> begin feeding the track Gst data with `rtcSendMessage`

The peer does get audio, I can hear it, but not before `rtcSendMessage` returns `RTC_ERR_FAILURE` several times and I see `... ERROR [121270] [{anonymous}::wrap@229] Track is closed` printed several times. What I guess is happening is Track needs some time to be open and I'm not waiting for some additional callback indicating the track is ready? Could you please point me to what that callback could be, or anything else I may be doing wrong?

**Comment by paullouisageneau** [comment]
You have to wait for the peer connection to connect and the track to open, the corresponding callback is `Track::onOpen`.
